R/call_map_api

```diff
@@ -0,0 +1,303 @@
+#' use for the binning of raster values. Should be between 10 and 50. If breaks
+#' is specified this values will be ignored. Default is 50.
+#' @param latMax maximum latitude of a rectancular bounding box restricting the
+#' search for species occurrences of the target group (optional).
+#' @param x numeric value to specify the longitudinal position of the raster
+#' extent. Default is 0.
+#' @param y numeric value to specify the latitudinal position of the raster
+#' extent. Default is 0.
+#' @param z numeric value to specify the zoom factor of the raster extent.
+#' Default is 0 for global extent.
+#' @author Jan Laurens Geffert, \email{laurensgeffert@@gmail.com}
+#' @details This function uses the arguments passed on to generate a query
+#' to the GBIF web map API. The API returns a web tile object that can be read
+#' as a png and converted into an R raster object. The break values or nbreaks
+#' generate a custom colour palette for the web tile, with each bin
+#' corresponding to one grey value. After retrieval, the raster is reclassified
+#' to the actual break values. This is a somewhat hacky but nonetheless
+#' functional solution in the absence of a GBIF raster API or an rgbif map API
+#' implementation.
+#' @references \url{http://en.wikipedia.org/wiki/List_of_Crayola_crayon_colors}
+#' @seealso \url{http://www.ecography.org/accepted-article/performance-tradeoffs-target-group-bias-correction-species-distribution-models}
+#' @keywords sampling bias, web map tile, grid, raster, GBIF
+#' @export
+
+# GENERATE MAP API CALL STRING -------------------------------------------------
+call_map_api <- function(
+  taxonkey = 1,
+  decades = c('NO_YEAR','PRE_1900','1900_1910','1910_1920','1920_1930','1930_1940','1940_1950','1950_1960','1960_1970','1970_1980','1980_1990','1990_2000','2000_2010','2010_2020'),
+  layers = c('OBS','SP','OTH'),
+  living = TRUE,
+  fossil = FALSE,
+  breaks = NULL,
+  nbreaks = NULL,
+  x = 0,
+  y = 0,
+  z = 0) {
+
+  # Check input arguments
+  if(!is.numeric(x) & !is.integer(x)) stop('x value supplied is not numeric.')
+  if(!is.numeric(y) & !is.integer(y)) stop('y value supplied is not numeric.')
+  if(!is.numeric(z) & !is.integer(z)) stop('z value supplied is not numeric.')
+  if(!is.numeric(taxonkey) & !is.integer(taxonkey)){
+    stop('value supplied for argument "taxonkey" is not numeric.')
+  }
+  # issues:
+  # - include spatial filter
+  # - include date filter
+  # - include functionality to use custom species list instead of taxon
+  # - add optional smoothing using spline?
+
+  if(exists('breaks') & !is.null(breaks)){
+    if(!is.numeric(breaks) & !is.integer(breaks)){
+      stop('break values supplied are not numeric.')
+    }
+  }
+  if(exists('nbreaks') & !is.null(nbreaks)){
+    if(!is.numeric(nbreaks) & !is.integer(nbreaks)){
+      stop('number of breaks value supplied is not numeric.')
+    }
+    if(nbreaks > 99){
+      stop('nbreaks value too large. Use a value between 1 and 99')
```

Comment:
I am not sure I understand this one correctly.
Are you suggesting to do it like this?
``` stop('nbreaks value too large. Use a value between 1 and 99', call. = FALSE) ```